server/users.ts

```diff
@@ -1018,20 +1018,22 @@ export class User extends Chat.MessageContext {
 	/**
 	 * @param updated the settings which have been updated or none for all settings.
 	 */
-	getUpdateuserText(...updated: string[]) {
+	getUpdateuserText(...updated: (keyof User)[]) {
 		const named = this.named ? 1 : 0;
-		const diff = {};
+		// This type is actually {-readonly [k in keyof User]?: User[k]}, but
+		// typescript dies if we use that. It should be safe enough as is, since
+		// all we do is stringify the value.
+		const diff: {-readonly [k in keyof User]?: any} = {};
 		const settings = updated.length ? updated : SETTINGS;
 		for (const setting of settings) {
-			// @ts-ignore - dynamic lookup
 			diff[setting] = this[setting];
 		}
 		return `|updateuser|${this.getIdentityWithStatus()}|${named}|${this.avatar}|${JSON.stringify(diff)}`;
 	}
 	/**
 	 * @param updated the settings which have been updated or none for all settings.
 	 */
-	update(...updated: string[]) {
+	update(...updated: (keyof User)[]) {
```

Comment:
Wait, why does `update` support partial updates in the first place? Did I approve this?

Reply:
apparently it was #5438

Reply:
`keyof User` seems way too broad, anyway; it should be a subset of `SETTINGS`.

I would do `SETTINGS = [...] as const`, and then `typeof SETTINGS[0]` or something?